actors/builtin/miner/miner_actor.go

```diff
@@ -282,13 +282,23 @@ func (a Actor) ProveCommitSector(rt Runtime, params *ProveCommitSectorParams) *a
 		rt.Abortf(exitcode.ErrNotFound, "no precommitted sector %v", sectorNo)
 	}
 
+	if rt.CurrEpoch() > precommit.PreCommitEpoch+MaxSealDuration[precommit.Info.RegisteredProof] || rt.CurrEpoch() < precommit.PreCommitEpoch+PreCommitChallengeDelay {
+		rt.Abortf(exitcode.ErrIllegalArgument, "Invalid ProveCommitSector epoch (cur=%d, pcepoch=%d)", rt.CurrEpoch(), precommit.PreCommitEpoch)
+	}
+
+	TODO()
+	// TODO HS: How are SealEpoch, InteractiveEpoch determined (and intended to be used)?
+	// Presumably they cannot be derived from the SectorProveCommitInfo provided by an untrusted party.
+
+	// will abort if seal invalid
```

Comment:
I think there's a rebase artefact here, HS completed and removed this TODO on master.